src/cli_main.cc

```diff
@@ -157,8 +158,14 @@ struct CLIParam : public XGBoostParameter<CLIParam> {
     if (name_pred == "stdout") {
       save_period = 0;
     }
-    if (dsplit == 0 && collective::IsDistributed()) {
-      dsplit = 2;
+    if (dsplit == 0) {
+      if (collective::IsFederated()) {
+        dsplit = 3;
```

Comment:
Can we use the enum symbol here instead of integer?

Reply:
Done.